### ISSUE TITLE
Inmutable?

### ISSUE THREAD

**Issue by BrodaNoel** [BrodaNoel/duix#18]
Check what happens if we mutate the state. And define what should happens (should it be immutable?)

For example:
‘’’js
const x = duix.get(‘x’);
x.foo = ‘bar’;
‘’’

Document it

**Comment by jehon** [comment]
In my own code, I did implement a "deepFreeze" of the objects...

```js
// Thanks to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze
function deepFreeze(object) {
    var propNames = Object.getOwnPropertyNames(object);
    for (let name of propNames) {
        let value = object[name];

        if (value && typeof value === "object") {
            deepFreeze(value);
        }
    }

    return Object.freeze(object);
}
```

**Comment by BrodaNoel** [comment]
@jehon I just saw that we broke something with the `deepEqual` check.

Check this code:

```js
const user = duix.get('user');
user.listOfSomething.push(123);
duix.set('user', user);
```

In line 2, we are mutating the state, so, when we do the PUSH value, now `user` have the `123` value. What happens here? Well... Then, we do the `duix.set`, but the `listOfSomething` already had the value, so, there is no change in the object, thus, no callback called.

The callback on `duix.subscribe` should receive a `newValue` that can be mutated with no mutating the value on the storage. it should retrieve a clone, or something like that.

How do you think we should handle it? I never used the `Object.freeze` approach.
Should we freeze the object before sending it to the callbacks?

**Comment by BrodaNoel** [comment]
@jehon I'm working on it

**Comment by BrodaNoel** [comment]
Fixed on `v3.0.0`
